lib/tapioca/dsl/compilers/active_record_relations.rb

```diff
@@ -679,6 +680,59 @@ def create_common_methods(common_relation_methods_module)
             end
           end
 
+          BATCHES_METHODS.each do |method_name|
+            case method_name
+            when :find_each
+              create_common_method(
+                "find_each",
+                common_relation_methods_module,
+                parameters: [
+                  create_kw_opt_param("start", type: "T.untyped", default: "nil"),
+                  create_kw_opt_param("end", type: "T.untyped", default: "nil"),
+                  create_kw_opt_param("batch_size", type: "Integer", default: "1000"),
+                  create_kw_opt_param("error_on_ignore", type: "T.untyped", default: "nil"),
+                  create_kw_opt_param("order", type: "Symbol", default: ":asc"),
+                  create_block_param("block", type: "T.nilable(T.proc.params(object: #{constant_name}).void)"),
+                ],
+                return_type: "T.nilable(T::Enumerable[#{constant_name}])",
+              )
+            when :find_in_batches
+              create_common_method(
+                "find_in_batches",
+                common_relation_methods_module,
+                parameters: [
+                  create_kw_opt_param("start", type: "T.untyped", default: "nil"),
+                  create_kw_opt_param("end", type: "T.untyped", default: "nil"),
```

Comment:
```suggestion
                  create_kw_opt_param("finish", type: "T.untyped", default: "nil"),
```

https://github.com/rails/rails/blob/bd11e520a2e81d2b272a9812ca21ae199d74bf16/activerecord/lib/active_record/relation/batches.rb#L102